pkg/controller/velerobackup/helpers_test.go

```diff
@@ -0,0 +1,149 @@
+package velerobackup
+
+import (
+	"reflect"
+	"testing"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
+	testclusterdeployment "github.com/openshift/hive/pkg/test/clusterdeployment"
+	testsyncset "github.com/openshift/hive/pkg/test/syncset"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	namespace = "notarealns"
+)
+
+var (
+	errDefault = errors.New("Not a real error")
+	errStatus  = kerrors.StatusError{
+		ErrStatus: metav1.Status{},
+	}
+)
+
+func assertHiveObjectsEqual(t *testing.T, expected, actual []*hiveObject) {
+	for i := range expected {
+		assert.Equal(t, expected[i].object, actual[i].object)
+	}
+}
+
+func getHashOfObjectsFuncReturnsError(objects ...interface{}) (string, error) {
+	return "", errDefault
+}
+
+func defaultChecksumFunc(object runtime.Object) string {
+	r := emptyReconcileBackup()
+
+	hobj, err := newHiveObject(object, r.logger)
+	if err != nil {
+		panic(err)
+	}
+
+	return hobj.checksum
+}
+
+func unchangedSyncSetBase() testsyncset.Option {
+	return func(syncSet *hivev1.SyncSet) {
+		syncSet.Name = "somesyncset-unchanged"
+		syncSet.Namespace = namespace
+		testsyncset.WithBackupChecksum(defaultChecksumFunc)(syncSet)
+	}
+}
+
+func changedSyncSetBase() testsyncset.Option {
+	return func(syncSet *hivev1.SyncSet) {
+		syncSet.Name = "somesyncset-changed"
+		syncSet.Namespace = namespace
+	}
+}
+
+func unchangedClusterDeploymentBase() testclusterdeployment.Option {
+	return func(clusterDeployment *hivev1.ClusterDeployment) {
+		clusterDeployment.Name = "someclusterdeployment-unchanged"
+		clusterDeployment.Namespace = namespace
+		testclusterdeployment.WithBackupChecksum(defaultChecksumFunc)(clusterDeployment)
+	}
+}
+
+func changedClusterDeploymentBase() testclusterdeployment.Option {
+	return func(clusterDeployment *hivev1.ClusterDeployment) {
+		clusterDeployment.Name = "someclusterdeployment-changed"
+		clusterDeployment.Namespace = namespace
+	}
+}
+
+func emptyReconcileBackup() *ReconcileBackup {
```

Comment:
The only use for this function is to get a logger to use. So, instead, have a func that returns the logger. This func can then be used in `fakeClientReconcileBackup`, too.